lib/cc/pull_requests.rb

```diff
@@ -1,8 +1,15 @@
 class CC::PullRequests < CC::Service
+  VALID_TOKEN_MESSAGE = "Access token is valid.".freeze
+  CANT_UPDATE_STATUS_MESSAGE = "Access token is invalid - can't update status.".freeze
+
   def receive_test
     setup_http
 
-    receive_test_status
+    if able_to_update_status?
+      { ok: true, message: VALID_TOKEN_MESSAGE }
+    else
+      { ok: false, message: CANT_UPDATE_STATUS_MESSAGE }
+    end
```

Comment:
I agree with you on this for the `GitHubPullRequests` functionality. See this [comment](https://github.com/codeclimate/codeclimate-services/pull/114#discussion_r77405047).

But the default behavior here only requires the token to be valid for setting commit statuses, not for commenting, so only one HTTP request. I think that the default behavior which uses the response for the error message is more helpful in this shared class.